### PR TITLE
Recognize 0 in customInitialValues

### PR DESCRIPTION
**Pull request by marwahaha**
fixes https://github.com/Strilanc/Quirk/issues/447

I noticed this was introduced at this commit: https://github.com/Strilanc/Quirk/commit/8e28c2a5b014615771afb9ab23630cffe0b0556b

The [INITIAL_STATES_TO_GATES](https://github.com/Strilanc/Quirk/blame/master/src/gates/AllGates.js#L357) list uses `undefined` instead of 0 to represent |0>, so passing in `0` to the function will not delete it from the `init: []` map.

I tested locally and this seems to fix it!

### CODE REVIEW

File: src/circuit/CircuitDefinition.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ class CircuitDefinition {
         if (newStateIndex !== undefined) {
             newVal = newStateIndex;
         }
-        if (newVal === undefined) {
+        if (newVal === undefined || newVal === 0) {
```
Comment (Strilanc) [inline src/circuit/CircuitDefinition.js:156]: You should add a unit test exercising this new code path in `CircuitDefinition.test.js`

**Review by Strilanc — CHANGES_REQUESTED** [review]
Looks good. Needs a test. Not sure why the CI isn't running... will check that out.

File: src/circuit/CircuitDefinition.js (hunk 1 above)
Comment (marwahaha) [reply src/circuit/CircuitDefinition.js:156]: I couldn't find a test that uses this method, so I added one.

**Review by Strilanc — APPROVED** [review]
Thanks!

**Pull request merged**